lib/UmdMainTemplatePlugin.js

```diff
@@ -124,16 +134,16 @@ class UmdMainTemplatePlugin {
 				) +
 				"	else if(typeof define === 'function' && define.amd)\n" +
 				(requiredExternals.length > 0 ?
-					(this.name && this.namedDefine === true ?
-						"		define(" + libraryName(this.name) + ", " + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n" :
+					(this.names.amd && this.namedDefine === true ?
+						"		define(" + libraryName(this.names.amd) + ", " + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n" :
 						"		define(" + externalsDepsArray(requiredExternals) + ", " + amdFactory + ");\n"
 					) :
-					(this.name && this.namedDefine === true ?
-						"		define(" + libraryName(this.name) + ", [], " + amdFactory + ");\n" :
+					(this.names.amd && this.namedDefine === true ?
+						"		define(" + libraryName(this.names.amd) + ", [], " + amdFactory + ");\n" :
```

Comment:
@sokra Thanks, I've added tests for `namedDefine` being `true` and `false`. I'm not sure why it was a string value before, as I'm not sure that's a valid value per the docs - I'd only copied that from other test cases.